src/components/form/HdCheckbox.vue

```diff
@@ -34,13 +35,15 @@
       @focus="handleFocus"
       @blur="handleBlur"
     >
-      <div class="checkbox__inner__box">
-        <div class="checkbox__inner__box__overlay"></div>
-        <div class="checkbox__inner__box__border"></div>
-        <HdIcon
-          :src="checkIcon"
-          class="checkbox__inner__box__tick"
-        />
+      <div class="checkbox__inner__box--shadow">
```

Comment:
`--shadow` here is not a modifier. It's more like a wrapper or container.
But looking at the component, BEM is being used as BEE...EEM 😄  (this is how we were doing it some time ago).
If fixing how BEM is being used in this component is out of this PR's scope, I'd suggest to go with the flow and use `__shadow` for now.

Reply:
Makes sense. 
I have just changed the class name for now, maybe fix BEM should be addressed in a future ticket.